gammapy/datasets/map.py

```diff
@@ -337,34 +337,53 @@ def data_shape(self):
         """Shape of the counts or background data (tuple)"""
         return self._geom.data_shape
 
-    def _energy_range(self, mask = None):
-        """Compute the energy range with or without the mask fit."""
-        energy = self._geom.axes["energy"].edges
-        energy_min, energy_max = energy[:-1], energy[1:]
+    def _energy_range(self, mask=None):
+        """Compute the energy range maps with or without the fit mask."""
+        geom = self._geom
+        energy = geom.axes["energy"].edges
+        idx = geom.axes.index_data("energy")
+        geom = geom.drop("energy")
 
         if mask is not None:
-            if mask.data.any():
-                mask = mask.data.any(axis=(1, 2))
+            mask = mask.data
+            if mask.any():
+                energy_shape = [1] * mask.ndim
```

Comment:
I think you could simplify the implementation here by relying on `np.argmax()`. See the following minimal example:
```python3
import numpy as np
from astropy import units as u

energy = [1, 3, 10] * u.TeV
data = np.ones((3, 2, 2), dtype=bool)
data[0, 0, 0] = False
data[:1, 1, 1] = False
data[:2, 0, 1] = False

idx = np.argmax(data, axis=0)
energy[idx]
```
`np.argmax()` will directly return the index of the first appearance. To get the upper energy range you could rely on `np.flip` like so:
```python3
data = np.flip(data, axis=0)
idx = np.argmax(data, axis=0)
energy[::-1][idx]
```